lib/Sema/ConstraintSystem.cpp

```diff
@@ -1448,9 +1452,33 @@ doesStorageProduceLValue(AbstractStorageDecl *storage, Type baseType,
     }
   }
 
-  // Unsettable storage decls always produce rvalues.
-  if (!storage->isSettableInSwift(useDC, base))
-    return false;
+  switch (storage->mutabilityInSwift(useDC, base)) {
+    case StorageMutability::Immutable:
+      // Immutable storage decls always produce rvalues.
+      return false;
+
+    case StorageMutability::Mutable:
+      break;
+
+    case StorageMutability::Initializable: {
+      // If the member is not known to be on the left-hand side of
+      // an assignment, treat it as an rvalue so we can't pass it
+      // inout.
+      if (!memberLocator)
+        break;
+
+      auto *anchor = getAsExpr(memberLocator->getAnchor());
```

Comment:
Should this me `simplifyToAnchor` instead?

Reply:
Yes, thanks!